src/Common/MorganStanley.ComposeUI.Common.VisualUtils/DispatcherAdapter.cs

```diff
@@ -0,0 +1,51 @@
+﻿/// ********************************************************************************************************
+///
+/// Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License").
+/// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+/// See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+/// Unless required by applicable law or agreed to in writing, software distributed under the License
+/// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+using Avalonia.Threading;
+using MorganStanley.ComposeUI.Common.Interfaces;
+using NP.Utilities.Attributes;
+
+
+namespace MorganStanley.ComposeUI.Common.VisualUtils
+{
+    /// <summary>
+    /// this class implements Avalonia UI thread marshalling functionality. 
+    /// </summary>
+    [Implements(typeof(ISyncContext), isSingleton:true)]
```

Comment:
What's this attribute?

Reply:
This is one of my main concerns - we shot ourselves in the foot with Unity because it was owned by a single person. I don't think we should go with a DI container nobody knows. Not because I don't trust it but because it's against the design principle of least dependencies. If we still go with this DI solution, we must provide adapters for plugins that want to use the de facto standard MSFT extension.

Reply:
I'll be happy to replace it with MEF, but standard MSFT has severe limitations and we should avoid using it.

Reply:
IoCy is better than MEF because it is built with MEF in mind. It does not have MEF complexity, it is multiplatform, tested on multiple platforms, open source and well documented on the codeproject. So, I prefer to keep it at least for now.

Reply:
> What's this attribute?

This attribute means that the ISyncContext is resolved to DispatcherAdapter by the Inversion of Control container.

Reply:
> This attribute means that the ISyncContext is resolved to DispatcherAdapter by the Inversion of Control container and it is resolved as a Singleton.

Can you please point me to the documentation for this DI container? I don't want to make premature comments without understanding the basics but to me having an `Implements` attribute sounds weird: if a class implements an interface then it should simply implement it, the abstraction-implementation mappings should live in the Composition Root.

Reply:
here is some documentation https://www.codeproject.com/Articles/1244087/Introducing-IoCy-Simple-and-Powerful-IoC-Container. It is my own container, so I am a bit partial. The good point through is that it is very small and every software guy or girl can become an expert in it within two hours. 
I am very open to other popular containers, but I think that Microsoft DependencyInjection container misses important functionality. 
ImplementsAttribute notifies that it should map the implementation type to the interface or base class mentioned as the first argument. So when the class is added to the container (especially if it is loaded dynamically) it will map into ISyncContext interface as a singleton and one will be able to obtain the object of the class by using 
`ISyncContext syncContext = container.Resolve<ISyncContext>();`